ERCS/erc-7715.md

```diff
@@ -0,0 +1,466 @@
+    data: any
+  }
+ 
+  permissions: {
+    type: string
+    data: any
+    required: boolean
+  }[]
+  
+  expiry: number,
+}
+```
+
+`signer` is an optional field that identifies the key or account associated with the session.  When not specified, it’s implied that the wallet will manage the session.  See the “Wallet-managed Sessions” section for details.
+
+`permissions` is a list of permissions.  See “Permission Types” section for details.
+
+Each permission can be either `required` or not.  If a permission is `required`, the wallet MUST grant the permission if it approves the request.  Otherwise, the wallet may reject the permission even if it approves the request (presumably because it grants other permissions).
+
+`expiry` is a UNIX timestamp (in seconds) that specifies the time by which this session MUST expire.
+
+
+Request example:
+
+```tsx
+{
+  signer: {
+    type: 'key',
+    data: {
+      id:'did:key:zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme',
+      }
+    }
+  permissions: [
+    {
+      type: 'erc20_spending_limit',
+      data: {
+        erc20Address: '0x...',
+        limit: '0x...',
+      }
+      required: true
+    },
+    {
+      type: 'gas_limit',
+      data: {
+        limit: '0x...',
+      },
+      required: false
+    }
+  ],
+  expiry: 1577840461
+}
+```
+
+#### Response Specification
+
+```tsx
+
+type IssuePermissionsResponse = {
+  grantedPermissions: {
+    type: string
+    data: any
+  }[]
+      
+  expiry: number
+  
+  signerData: {
+    userOpBuilder?: `0x${string}`
+    managerAddress?: `0x${string}`
+  },
+  factory?: `0x${string}`,
+  factoryData?: string,
+	
+  permissionsContext: string
+}
+```
+
+- `grantedPermissions` is a list of permissions that were granted.  The wallet MUST grant `required` permissions if the request was approved.
+- `expiry` is the expiry time that was granted.  The wallet MAY grant a different expiry time than the one that the DApp requested.
+- `factory` and `factoryData` are as defined in [ERC-4337](./eip-4337.md).  They are either both specified, or none.  If the account has not yet been deployed, the wallet MUST return `factory` and `factoryData`, and the DApp MUST deploy the account by calling the `factory` contract with `factoryData` as the calldata.
+- `permissionsContext` is an opaque string that identifies the session. It MAY be the `context` as defined in [ERC-7679](./eip-7679.md) and [ERC-7710](./eip-7710.md).  See “Rationale” for details.
+
+If the request is malformed or the wallet is unable/unwilling to issue permissions, wallet MUST return an error with a code as defined in [ERC-1193](./eip-1193.md).
+
+`wallet_issuePermissions` response example:
+
+```tsx
+{
+  grantedPermissions: [
+    {
+      type: 'erc20_spending_limit',
+      data: {
+        erc20Address:'0x...',
+        limit: '0x...',
+      },
+    },
+    {
+      type: 'gas_limit',
+      data: {
+        limit: '0x...',
+      },
+      required: false
+    }
+  ],
+  
```

Comment:
would it make sense here to add signerdata and factory+data to show all the main fields?

Reply:
but technically these are optionals and signerData has both a 7679 and a 7710 scenario

so we would need separate examples for each